### PR TITLE
fix: cover all MetadataLastModified test case

### PR DESCRIPTION
**Pull request by lucarin91**
#### :question: What

This is an update of PR #131. It adds the test case in which the date is set but it is zero.

#### :white_check_mark: Checklists

- [x] Has `make gofmt-write` been run on the code?
- [x] Has `make govet` been run on the code? Has the code been fixed accordingly to the output?
- [x] Have the changes been added to the [CHANGELOG.md](./CHANGELOG.md) file?
- [x] Have the steps in [CONTRIBUTING.md](./CONTRIBUTING.md) been followed to update a Go module?

### CODE REVIEW

File: filter/metadata_lastmodified_test.go [hunk 1]
```diff
@@ -9,22 +9,32 @@ import (
...
 			name:         "time set",
-			lastmodified: time.Date(2009, time.November, 10, 23, 0, 0, 0, time.UTC),
+			lastmodified: ptr(time.Date(2009, time.November, 10, 23, 0, 0, 0, time.UTC)),
 			dst:          "f2",
 			want:         "1257894000",
 		},
 		{
-			name:         "time empty",
-			lastmodified: time.Time{},
+			name:         "time set at zero",
```
Comment (arl) [inline filter/metadata_lastmodified_test.go:30]: nit: better to cal this `zero-value` as it's Go name for a default-initialized value, moreover it's not set to zero explicitely

File: filter/metadata_lastmodified_test.go [hunk 2]
```diff
@@ -9,22 +9,32 @@ import (
...
-		lastmodified time.Time
+		lastmodified *time.Time
 		dst          string
 		want         string
 		wantErr      bool
 	}{
 		{
 			name:         "time set",
-			lastmodified: time.Date(2009, time.November, 10, 23, 0, 0, 0, time.UTC),
+			lastmodified: ptr(time.Date(2009, time.November, 10, 23, 0, 0, 0, time.UTC)),
```
Comment (arl) [inline filter/metadata_lastmodified_test.go:25]: you could also directly use:
```go
ptr(time.Unix(1257894000, 0))
```

**Review by arl — APPROVED** [review]
Looks good to me.
I let only 2 naming comments

**Review by arl — APPROVED** [review]
2nd review: OK

**Comment by lucarin91** [comment]
As suggested by @tommyblue in #135 we can add additional information in the filter documentation about the field `MetadataLastModified` in the Metadata. Is there any documentation about the Record Metadata?

**Pull request merged**